ibis/expr/rules.py

```diff
@@ -294,12 +295,35 @@ def _promote_integral_binop(exprs, op):
     return dt.highest_precedence(dtypes)
 
 
+def _infer_decimal_dtype(args, op):
+    # TODO: Add support for the maximum precision and maximum scale
+    # TODO: Currently assumes that there are only two args. Can there be more?
+    if op is operator.mul:
+        return dt.Decimal(
+            args[0].type().precision + args[1].type().precision,
+            args[0].type().scale + args[1].type().scale,
+        )
+    if op is operator.add or op is operator.sub:
+        return dt.Decimal(
+            max(
+                args[0].type().precision - args[0].type().scale,
+                args[1].type().precision - args[1].type().scale,
+            )
+            + max(args[0].type().scale, args[1].type().scale)
+            + 1,
+            max(args[0].type().scale, args[1].type().scale),
+        )
+    return highest_precedence_dtype(args)
+
+
 def numeric_like(name, op):
     @immutable_property
     def output_dtype(self):
         args = getattr(self, name)
         if util.all_of(args, ir.IntegerValue):
             result = _promote_integral_binop(args, op)
+        elif util.all_of(args, ir.DecimalValue):
+            result = _infer_decimal_dtype(args, op)
```

Comment:
Can you add a test that hits this code path?

This can go in `ibis/tests/expr/test_decimal.py`.